experiments/SlabLand/README.md

```diff
@@ -0,0 +1,161 @@
+
+2. Flux accumulation
+    - In order to integrate fluxes in time consistently with the Atmos time stepper, we define a new prognostic variable `F_ρθ_accum` whose source function accumulates the fluxes, so that $F\_ρθ\_accum = \int F_{tot} dt$
+
+            @inline function source!(atmos_model::ModelSetup, _...)
+                source.F_ρθ_accum = calculate_land_sfc_fluxes(atmos_model, _...) 
+            end
+
+3. Flux / state storage and communication
+    - the coupler stores 2 fields:
+        - `EnergyFluxAtmos` which collects `F_ρθ_accum / Δt` calculated and accumulated in Atmos (with $Δt$ denoting the period of the coupling cycle), and is called by land for its $T_{sfc}$ equation
+        - `LandSurfaceTemperature` which collects the state of the slab Land model, namely $T_{sfc}$ and is called by atmos when calculating surface fluxes
+
+
+## **Tests**
+
+- Dry heat diffusion
+    - Simplifications
+        - $g_a$ is constant
+        - (μ, ν, $\kappa$) = (0, 0, 1e-5)
+        - $\vec{u} = (0,0,0)$
+        - $G = 0$ and $F_a = 0$
+    - [results](https://docs.google.com/document/d/1JKK8wFKPq3Jo3D4flXZiY3WAUPqYE19pXRvwJgpDwjw/edit)
+- Dry advection-diffusion
+    - Simplifications
+        - $g_a$ is constant
+        - (μ, ν, $\kappa$) = (0, 0, 1e-5)
+        - $\vec{u} = (1,0,0)$
+        - $G \neq 0$ and $F_a = 0$
+    - [results](https://docs.google.com/document/d/1JKK8wFKPq3Jo3D4flXZiY3WAUPqYE19pXRvwJgpDwjw/edit)
+
+## **Pipeline** (probably delete)
+
+For this simple implementation, we assume:
+-  a dry setup 
+- atmosphere only governed by heat diffusion
+- aerodynamic conductance (i.e. $g_a = |u| c_D$, with $c_D$ representing the drag coefficient) is constant . Using explicit time stepping the formulation is implemented as follows:
+
+
+1) Coupler sends its field `LandSurfaceTemperature` and transforms it into Atmos field `auxiliary.T_sfc`
+2) Atmos initializes its state according to the user-defined ICs
+3) Atmos performs all its timesteps within the coupling cycle, with the boundary conditions at the coupled boundary setting the total normal flux to be equal to the flux coming from the land (`fluxᵀn.ρθ = - F_tot / cp_d`, so that $\partial_t \rho \theta = \nabla_z \sdot F_{tot}/cp_d$)
+4) The same flux `F_tot` is calculated in the `source!` function and saved as `state.F_ρθ_accum`, which ensures that the flux will be integrated and accumulated in time, consistent with the Atmos time stepping. 
+5) Coupler converts Atmos field `state.F_ρθ_accum` to the coupler field `EnergyFluxAtmos` 
+6) Coupler sends its field `EnergyFluxAtmos` and transforms it into Land field `auxiliary.F_ρθ_prescribed`
+5) Land performs its own initialization and physics (`G`), which is then added to the  `auxiliary.F_ρθ_prescribed` (corresponding to $F_{tot}$ above) in the `source!` function to solve:
+$$T_s^{n+1} = T_s^n +\frac{F_{tot}^{n...}}{\rho_s c_s h_s}\Delta t$$
```

Comment:
i also found this pipeline really helpful. 

the use of F_tot is inconsistent here - in line 87 it refers to the non G fluxes, here it refers to G + non G fluxes.

Reply:
I also wonder if using the subscript n with F_tot is correct? If Im understanding, it does use land values from the previous land step, but it's an accumulation of the flux (using atmos values at intermediate t) over the whole land step. Might be worth stating that?

If one wanted to use a different explicit stepper on the land side (not forward Euler), is that possible? Would the "coupling cycle" dt be e.g. the different dts for the RK stages?

Reply:
@kmdeck unless I'm mistaken, yes, the choice of the land timestepper will determine how $T_s^{n+1}$ is computed, and this does not have to be forward Euler, of course.

The "coupling cycle" refers to the synchronization timescale of the two solvers. At the beginning/end of the coupling cycle, coupled field information is exchanged. Within that coupling cycle, the individual components can take as many substeps as they need, but they don't exchange information.

Reply:
thanks! @jb-mackay. I'll think about it more! May still follow up with ?s, but that sounds good.